src/System.Slices/System/Buffers/Memory.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime;
+
+namespace System.Buffers
+{
+    /// <summary>
+    /// Represents a Span<byte> factory</byte>
+    /// </summary>
+    /// <remarks>
+    /// This struct is not safe for multithreaded access, i.e. it's subject to struct tearing that can result in unsafe memory access. 
```

Comment:
Does it need to be safe? We could make it an unsafe type that internals of the pipeline use very carefully. Then user code always gets spafe Spans.  
To me the reason for this type to be a struct is that the main usage will be to embed it in some "pipeline context" and it will just cost additional allocation without much improved safety.